src/map/mercenary.cpp

```diff
@@ -474,137 +443,467 @@ void mercenary_kills(struct mercenary_data *md){
+
+		if (!this->asUInt16(node, "ElementLevel", level))
+			return 0;
+
+		if (!CHK_ELEMENT_LEVEL(level)) {
+			this->invalidWarning(node["ElementLevel"], "Invalid mercenary element level %hu, defaulting to 1.\n", level);
+			level = 1;
+		}
+
+		mercenary->status.ele_lv = static_cast<uint8>(level);
+	} else {
+		if (!exists)
+			mercenary->status.ele_lv = 1;
+	}
+
+	if (this->nodeExists(node, "WalkSpeed")) {
+		uint16 speed;
+
+		if (!this->asUInt16(node, "WalkSpeed", speed))
+			return 0;
+
+		if (speed < MIN_WALK_SPEED || speed > MAX_WALK_SPEED) {
+			this->invalidWarning(node["WalkSpeed"], "Invalid mercenary walk speed %hu, capping...\n", speed);
+			speed = cap_value(speed, MIN_WALK_SPEED, MAX_WALK_SPEED);
+		}
+
+		mercenary->status.speed = speed;
+	} else {
+		if (!exists)
+			mercenary->status.speed = DEFAULT_WALK_SPEED;
+	}
+
+	if (this->nodeExists(node, "AttackDelay")) {
+		uint16 speed;
+
+		if (!this->asUInt16(node, "AttackDelay", speed))
+			return 0;
+
+		mercenary->status.adelay = cap_value(speed, 0, 4000);
+	} else {
+		if (!exists)
+			mercenary->status.adelay = 4000;
+	}
+	
+	if (this->nodeExists(node, "AttackMotion")) {
+		uint16 speed;
+
+		if (!this->asUInt16(node, "AttackMotion", speed))
+			return 0;
+
+		mercenary->status.amotion = cap_value(speed, 0, 2000);
+	} else {
+		if (!exists)
+			mercenary->status.amotion = 2000;
+	}
+
+	if (this->nodeExists(node, "DamageMotion")) {
+		uint16 speed;
+
+		if (!this->asUInt16(node, "DamageMotion", speed))
+			return 0;
+
+		mercenary->status.dmotion = speed;
+	} else {
+		if (!exists)
+			mercenary->status.dmotion = 0;
+	}
+
+	mercenary->status.aspd_rate = 1000;
+
+	if (this->nodeExists(node, "Skills")) {
+		const YAML::Node &skillsNode = node["Skills"];
+
+		for (const YAML::Node &skill : skillsNode) {
+			std::string skill_name;
+
+			if (!this->asString(skill, "Name", skill_name))
+				return 0;
+
+			uint16 skill_id = skill_name2id(skill_name.c_str());
+
+			if (skill_id == 0) {
+				this->invalidWarning(skill["Name"], "Invalid skill %s, skipping.\n", skill_name.c_str());
+				return 0;
+			}
+
+			if (!SKILL_CHK_MERC(skill_id)) {
+				this->invalidWarning(skill["Name"], "Skill %s (%u) is out of the mercenary skill range [%u-%u], skipping.\n", skill_name.c_str(), skill_id, MC_SKILLBASE, MC_SKILLBASE + MAX_MERCSKILL - 1);
+				return 0;
+			}
+
+			uint16 level;
+
+			if (!this->asUInt16(skill, "MaxLevel", level))
+				return 0;
+
+			if (level == 0) {
+				if (mercenary->skill.erase(skill_id) == 0)
+					this->invalidWarning(skill["Name"], "Failed to remove %s, the skill doesn't exist for mercenary %hu.\n", skill_name.c_str(), id);
+				continue;
+			}
+
+			mercenary->skill[skill_id] = level;
+		}
+	}
+
+	if (!exists)
+		this->put(id, mercenary);
+
+	return true;
 }
 
 /**
 * Init Mercenary datas
 **/
 void do_init_mercenary(void){
-	mercenary_readdb();
-	mercenary_read_skilldb();
+	mercenary_db.load();
 
 	//add_timer_func_list(mercenary_contract, "mercenary_contract");
```

Comment:
```suggestion
	add_timer_func_list(merc_contract_end, "merc_contract_end");
```

Reply:
`add_timer_func_list` was originally commented out, is that intended?

Reply:
Seems like someone did not know what he was doing... :-)
I just commented because I saw it now during review.